### PR TITLE
Added category and summary metadata to documentation files

### PR DESCRIPTION
**Pull request by nderwin**
Added category and summary metadata to documentation files

Signed-off-by:Nathan Erwin <nathan.d.erwin@gmail.com>

### CODE REVIEW

File: docs/src/main/asciidoc/all-builditems.adoc [hunk 1]
```diff
@@ -7,7 +7,8 @@ and pull requests should be submitted there:
 https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 ////
 include::./attributes.adoc[]
-
+:categories: writing-extensions
+:summary: Explore all the BuildItems you can consume/produce in your extensions.
 = Build Items
```
Comment (ebullient) [inline docs/src/main/asciidoc/all-builditems.adoc:12]: The title here needs to move up to line 9, doesn't it?

File: docs/src/main/asciidoc/all-config.adoc [hunk 2]
```diff
@@ -7,7 +7,7 @@ and pull requests should be submitted there:
 https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 ////
 = All configuration options
-
 include::./attributes.adoc[]
-
+:categories: core
+:summary: List all the configuration properties per extensions
```
Comment (ebullient) [inline docs/src/main/asciidoc/all-config.adoc:12]:
```suggestion
:summary: List all the configuration properties per extensions

```

**Review by ebullient — CHANGES_REQUESTED** [review]
A few adjustments. Thank you so much for tackling this. AMAZING.

File: docs/src/main/asciidoc/all-builditems.adoc (hunk 1 above)
Comment (nderwin) [reply docs/src/main/asciidoc/all-builditems.adoc:12]: It doesn't help with the index file generation, but sure, for consistency with the rest of the files.

File: docs/src/main/asciidoc/all-config.adoc (hunk 2 above)
Comment (nderwin) [reply docs/src/main/asciidoc/all-config.adoc:12]: Will do.

**Review by ebullient — APPROVED** [review]

**Pull request merged**